scripts/init-submodules-no-riscv-tools-nolog.sh

```diff
@@ -74,6 +74,8 @@ CHIPYARD_DIR="$(dirname "$DIR")"
 
 cd "$CHIPYARD_DIR"
 
+source $CHIPYARD_DIR/scripts/conda-activate-env.sh
+
```

Comment:
Why does this need to be here? To get the most recent git version?

I would prefer that this script still works without `conda`. Just tell the user in docs to activate the conda env before running anything else in Chipyard.

Also I would prefer that this doesn't activate conda in the env.sh script.

So the documented flow for users should be 
1. Install Conda
2. conda-env-setup.sh
3. conda-activate-env.sh
4. Do all other chipyard stuff (including init submodules).

And power users or millennium users can skip 1-3.

Reply:
I get the feeling that if we don't explicitly call conda activate in our scripts, avg. users will forget about this and will get weird errors that should be solved with conda activated. For now, I will remove these explicit calls, but I'll add a warning if the environment isn't set up like the check for the hash (aka you need to answer the prompt to continue).

Reply:
What tends to break in `init-submodules` without the conda-managed env? If the issue is the git version is too old, the script can check that and error with a message like.
```
ERROR: Git version {version} <= {supported version}. Did you create and activate the chipyard conda environment?
```

The prompt in build-toolchains is fine.

Reply:
I agree with Jerry.  You should decouple setting up the environment from running the flow.  Ask users to supply you with the output of `env | sorted` whenever they submit an issue that should be enough to tell you whether they are on a mill machine and whether they have conda activated or not (look for CONDA_PREFIX or CONDA_DEFAULT_ENV).